src/pydicom/filereader.py

```diff
@@ -155,11 +154,19 @@ def data_element_generator(
             # defend against switching to implicit VR, some writer do in SQ's
             # issue 1067, issue 1035
 
-            if not (b"AA" <= vr <= b"ZZ") and config.assume_implicit_vr_switch:
+            if vr in ENCODED_VR:  # try most likely solution first
+                vr = vr.decode(default_encoding)
+                if vr in EXPLICIT_VR_LENGTH_32:
+                    bytes_read = fp_read(4)
+                    length = extra_length_unpack(bytes_read)[0]
+                    if debugging:
+                        debug_msg += " " + bytes2hex(bytes_read)
+            elif not (b"AA" <= vr <= b"ZZ") and config.assume_implicit_vr_switch:
                 # invalid VR, must be 2 cap chrs, assume implicit and continue
                 vr = None
-                group, elem, length = implicit_VR_struct.unpack(bytes_read)
+                group, elem, length = implicit_VR_unpack(bytes_read)
             else:
+                # Possibly an unknown VR, mostly likely implicit
                 vr = vr.decode(default_encoding)
                 if vr in EXPLICIT_VR_LENGTH_32:
                     bytes_read = fp_read(4)
```

Comment:
If unknown VR then can't actually be in `EXPLICIT_VR_LENGTH_32`, so will always be treated as have a 2 byte length field.